src/app/components/shared/date-time-filter/filter.component.ts

```diff
@@ -0,0 +1,199 @@
+import { AfterViewInit, Component, EventEmitter, Input, Output, ViewChild } from "@angular/core";
+import { NgForm } from "@angular/forms";
+import { Comparisons, DateExpressions, Filters, InnerFilter } from "@baw-api/baw-api.service";
+import { isInstantiated } from "@helpers/isInstantiated/isInstantiated";
+import { withUnsubscribe } from "@helpers/unsubscribe/unsubscribe";
+import { Project } from "@models/Project";
+import { Region } from "@models/Region";
+import { Site } from "@models/Site";
+import { NgbCalendar, NgbDate, NgbDateParserFormatter } from "@ng-bootstrap/ng-bootstrap";
+import { debounceTime, distinctUntilChanged, takeUntil } from "rxjs";
+import { defaultDebounceTime } from "src/app/app.helper";
+
+export interface Model {
+  projects?: Project[];
+  regions?: Region[];
+  sites?: Site[];
+  dateFiltering?: boolean;
+  dateStartedAfter?: NgbDate;
+  dateFinishedBefore?: NgbDate;
+  todFiltering?: boolean;
+  todIgnoreDst?: boolean;
+  todStartedAfter?: string;
+  todFinishedBefore?: string;
+}
+
+@Component({
+  selector: "baw-date-time-filter",
+  templateUrl: "filter.component.html",
+  styleUrls: ["filter.component.scss"]
+})
+export class DateTimeFilterComponent extends withUnsubscribe() implements AfterViewInit {
+  public constructor(
+    private formatter: NgbDateParserFormatter,
+    private calendar: NgbCalendar
+  ) { super(); }
+
+  @ViewChild(NgForm) public form: NgForm;
+
+  @Input() public project: Project;
+  @Input() public region: Region;
+  @Input() public site: Site;
+  @Output() public constructedFilters = new EventEmitter<Filters>();
+
+  private previousFilters: string;
+  public model: Model = { todIgnoreDst: true };
+  public errors: {
+    todBoundaryError?: boolean;
```

Comment:
`todBoundaryError` is set in the `setTimeOfDayFilter()` method and used as a predicate whether the start date is < than the end date (out of bounds date)

![image](https://user-images.githubusercontent.com/33742269/222031582-763249a1-0d2d-4d2a-8f96-af3b3e32f378.png)

If the error `todBoundaryError` condition is true, the following error is emitted to the user.
![image](https://user-images.githubusercontent.com/33742269/222031753-c4e70bb9-fa4f-491e-865b-6126ec0409e5.png)

In retrospective, we should either move this to a getter method (similar to how the date bound are validated), or add the date bound validations to this `errors` object (so that there are not two separated ways to fetch errors from a filter component).

In my opinion, exporting an errors interface and adding the date boundary & format errors to this object is the most "correct" implementation and may prove useful in the future.

I will await your opinion though

Reply:
I am commenting on the naming. Suggested name: `timeOfDayStartNotLessThanEnd`. There are 6 other boundary conditions I can think of and the old name could refer to any of them.

But yeah,  like I said elsewhere kinda weird to see this here, I would have expected to see these entered as validations in the form.